Adding Stm32u5 documentation (#68)
AVSystem/developer-zone: docs/Anjay_integrations/Anjay_ST_integration/B-U585I-IOT02A-BG96.md

   ```

#### Step 2: Compiling the board

0. Connect the STM32U585I-IOT02A board to a USB port of your machine.
0. Go to the STM32CubeIDE.
0. Import the project cloned in the previous step to your workspace:
    - From the navigation bar, select **File** and click **Import**.
    - From the **General** list, select **Existing Projects into Workspace** and click **Next**.
    - In **Select root directory**, indicate the catalog containing the cloned Anjay freeRTOS client repository.
    - In the **Projects** field, select **Anjay-freertos-client-B-U585I-IOT02A-BG96** and click **Finish**.
    ![Import project](images/import.PNG "Import project")
0. In the Project Explorer, navigate to the **Anjay-freertos-client-B-U585I-IOT02A-BG96** project:
    - Right-click on the project name and select **Build Project**. The build should take less than one minute to complete.
    - After the build is finished, right-click on the project name, select **Run As** and click the **1 STM32 Cortex-M C/C++ Application** option.
        - In the **Lauch Configuration Selection**, choose the **Anjay-freertos-client-B-U585I-IOT02A-BG96** option and click **OK**.

> 0. After the build and run are complete, the board is now compiled.

Reviewer: ```suggestion
0. After the build and run steps are done, the board is now flashed and ready for next steps.
```
Or something like that. `the board is now compiled` is not a correct statement.

## Connecting to the LwM2M Server

> To connect to Coiote IoT Device Management LwM2M Server, please register at https://eu.iot.avsystem.cloud/.

Reviewer: ```suggestion
To connect to Coiote IoT Device Management LwM2M Server, please register at https://eu.iot.avsystem.cloud.
```
A hyperlink maybe?

!!! note

>     If you use BG96-based configuration, you must upgrade the firmware of the modem to at least the `BG96MAR02A08M1G` revision. Older versions may cause unexpected loss of connection.

Reviewer: ```suggestion
    If you use BG96-based configuration, you must upgrade the firmware of the modem to at least `BG96MAR02A08M1G` revision. Older versions may cause unexpected loss of connection.
```

To connect the board:

1. Log in to Coiote DM and from the left side menu, select **Device Inventory**.
2. In **Device Inventory**, click **Add device**.
3. Select the **Connect your LwM2M device directly via the Management server** tile.
       ![Add via Mgmt](images/mgmt_tile.png "Add via Mgmt")
    3. In the **Device credentials** step:
         - In the **Device ID** enter your board endpoint name, e.g. `test_device`.
             ![Device credentials step](images/add_mgmt_quick.png "Device credentials step")
         - In the **Security mode** section, select the **PSK (Pre-Shared Key)** mode:
              - In the **Key identity** field, type the same name as in the `Endpoint name` field
              - In the **Key** field, type the shared secret used in the device-server authentication.
    4. Click the **Add device** button and **Confirm** in the confirmation pop-up.

>     5. In the **Connect your device** step, wait for the board to connect.
>     6. Click **Go to device** to see your added device dashboard.

Reviewer: ```suggestion
    5. In the **Connect your device** step, click **Go to device** to see your added device dashboard.
```
the board won't connect, because you haven't set the board's credentials yet

    !!! tip

>         LwM2M Server URI, endpoint name and other informations you can be found in the **configuration** tab.

Reviewer: I would move it to the `Configuring the Client -> Press any key and in the configuration menu...` section, where you actually must set board's credentials.

## Configuring the Client

1. With the board still connected to a serial port interface, open a serial communication program.
2. Press the reset button located on the board. This should trigger the following prompt:

    ``Press any key in 3 seconds to enter config menu...``

3. Press any key and in the configuration menu, change the default credentials to your data by following the instructions presented in the program and save it.

>    ![Client configuration](images/config_menu1.png "Client configuration"){:style="float: left;margin-right: 1177px;margin-top: 17px;margin-bottom: 17px;"}

Reviewer: I would change/update the screenshot, in the screenshot e.g. endpoint_name is `michal-freertos`, but earlier in the tutorial in the Coiote DM we have `test_device`, that can confuse some people.

# B-U585I-IOT02A/BG96

Integrate your B-U585I-IOT02A Discovery kit board along with the default-provided Quectel BG96 modem.

## Prerequisites

- The STM32U585I-IOT02A/BG96 board with a USB cable.

> - Installed **STM32CubeIDE**.

Reviewer 2: ```suggestion
- Optional: Installed **STM32CubeIDE**.
```

- Installed **minicom** (for Linux) or RealTerm or PuTTy (for Windows) or other serial communication program.
- A user with access to the Coiote IoT Device Management platform.

> ## Prepare binaries

Reviewer 2: For what I gather here, we have two options to flash the board: using ready-made binaries or build the project in STM32CubeIDE. If so, I'd emphasise this by adding `### Option 1: Use already built binary` and `### Option 2: Start developement using samples` to the section titles.

Author: @mkrawczyk-avs I prefer to use note before the optional part. For me, it looks better on page.

> ### Use an already built binary

Reviewer 2: ```suggestion
### Option 1: Use an already built binary
```

To get the latest binaries, navigate to [Anjay-freertos-client](https://github.com/AVSystem/Anjay-freertos-client/releases/).

In this case we are going to use the Anjay-freertos-client-B-U585I-IOT02A-BG96.bin file which after being downloaded must be added to STM32U585I-IOT02A/BG96 board folder.

To flash the board, drag the .bin file to B-U585I-IOT02A device which you can find in **devices and drives** part in **This PC**. You will see the blinking diode on your board. As soon as the diode stops blinking, the flashing has finished.

Now you can go to the [Connecting to the LwM2M Server part ](#connecting-to-the-lwm2m-server)

> ### Start developement using samples

Reviewer 2: ```suggestion
### Option 2: Start development using samples
```
